AmazonWeb1/src/main/java/com/amazon/servlet/AuthenticationServlet.java

```diff
@@ -0,0 +1,116 @@
+package com.amazon.servlet;
+
+import com.amazon.controller.AuthenticationController;
+import com.amazon.exception.ServletException;
+import com.amazon.model.User;
+import org.json.JSONObject;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+/**
+ * <p>
+ * Represents the servlet request and response for {@link User} SignIn
+ * </p>
+ *
+ * @author Roshan B
+ * @version 1.0
+ */
+@WebServlet ("/authentication")
+public class AuthenticationServlet extends HttpServlet {
+
+    private final AuthenticationController authenticationController;
+
+    public AuthenticationServlet() {
+        authenticationController = AuthenticationController.getInstance();
+    }
+
+    /**
+     * <p>
+     *     Represent the post method in {@link HttpServlet} to provide {@link User} sign up
+     * </p>
+     * @param request Represents the request object from the api
+     * @param response Represents the response object sent from this servlet method
+     */
+    public void doPost(final HttpServletRequest request, final HttpServletResponse response) {
+
+        try {
+
+            final StringBuffer buffer = new StringBuffer();
+            String value;
+
+            while ((value = request.getReader().readLine()) != null) {
```

Comment:
What is the time complexity to get the reader here?

Reply:
order of (1).